eos_data_distribution/ndn/http.py

```diff
@@ -99,25 +111,31 @@ def _got_stream(self, msg, task, data):
 
     def _got_buf(self, data, buf):
         data.setContent(buf)
-        self.sendFinish(data)
+        self.emit('data', data)
+        logger.debug ('getter: got buf')
 
+class Producer(chunks.Producer):
+    def __init__(self, name, url, session=None, *args, **kwargs):
+        super(Producer, self).__init__(name, cost=DEFAULT_COST, *args, **kwargs)
+        self._getter = Getter(url, session=session, chunk_size=self.chunk_size)
+        self._getter.connect('data', lambda o, d: self.sendFinish(d))
+
+    def _get_final_segment(self):
+        return self._getter._size // self.chunk_size
+
+    def _send_chunk(self, data, n):
+        self._getter.soup_get(data, n)
 
 if __name__ == '__main__':
-    from gi.repository import GLib
     import re
+    from . import test
```

Comment:
I think these would be better as `unittest`s, so they can be picked up by setuptools and autopkgtest.